69.md

```diff
@@ -0,0 +1,293 @@
+NIP-69
+======
+
+Nostr Offer STRings
+-------------------
+
+`rfc` `optional`
+
+This NIP proposes a format for static payment codes in Nostr as a successor to LNURL-Pay, enabling users to initiate Lightning Network payments by scanning or clicking a string.
+
+## Motivation
+
+Reliance on LNURL has led to centralization via custodial solutions due to the legacy baggage of IP4-NAT/Domain/SSL requirements. Nostr's use-cases are already centered around Lightning, whether as a wallet connector, zap receipts, or its overlapping network effects, rendering Nostr as a de facto "3rd layer" for Lightning and a natural successor to LNURL.
+
+When layered over kind 21000, NWC, or other potential Lightning RPCs, this specification enables a seamless user experience similar to legacy LNURL-Pay, without the domain and SSL requirements. Additionally, the signed nature of Nostr communications minimizes the trust requirements inherent in LNURL when a receiving node must outsource the serving of web requests.
+
+## Specification
+
+### Static Payment Code Format
+
+The static payment code is a bech32 (per [NIP-19](19.md)) encoded string prefixed with `noffer`. The encoded string will include the following TLV (Type-Length-Value) items:
+
+- `0`: The 32 bytes of the receiving service's public key, encoded in hex.
+- `1`: The relay URL where the receiving service subscribes to payment requests.
+- `2`: The offer identifying string.
+- `3`: A flag indicating the pricing type:
+  - `0`: Fixed price
+  - `1`: Variable price
+  - `2`: Spontaneous payment (payer specifies the amount)
+- `4`: The price in sats (optional for display purposes).
+
+If neither the price nor the pricing type flag is present, the sender may assume it is a spontaneous payment offer.
+
+Example static payment code structure:
+
+```
+noffer1...
+  0: <receiver_public_key_in_hex>
+  1: <relay_url>
+  2: <offer_id_string> 
+  3: <pricing_type_flag> (optional, 0 for fixed, 1 for variable, 2 for spontaneous)
+  4: <price_in_sats> (optional)
```

Comment:
As a future extension, it would be awesome to think about specifying amounts in other currencies (although payment is still over lightning). Bolt-12 has a mechanism for this and LNURL has [LUD-21](https://github.com/lnurl/luds/pull/251) which has some decent usage now.

Reply:
Agreed, I thought through a bit already but haven't actually worked through an implementation...

Thinking is doable in this version, you would make a variable price offer, set a `price_usd` (or whatever unit) as an optional value, the service would calculate that on invoice generation, and then the wallet could take the response and check exchange rates to ensure its in the ballpark of what the offer had quoted

Reply:
Added reference to fiat units in relation to variable pricing